Docs/development/r-analyses-guide.md

```diff
@@ -122,54 +122,19 @@ We have to determine early in the analysis whether we are ready to compute the r
 </details>
 
 ## Step 3 - Reading the Dataset
-After selecting a dataset in JASP, you still have to load it into your R analysis. Reading data is always a tricky matter. Computers are devoid of common sense, so very often we have to explicitly specify how variables must be interpreted (are they integers?, characters?, ...). We recommend you to use our auxiliary function `.readDataSetToEnd()`, with the following optional arguments.
+By default, recent versions of JASP immediately load your dataset into your R analysis.
 
-- `columns`: columns that must be read without specific coercing to a datatype
-- `columns.as.numeric`: columns that must be read as numeric
-- `columns.as.ordinal`: columns that must be read as ordinal
-- `columns.as.factor`:  columns that must be read as factor
-- `all.columns`: boolean specifying if the entire dataset should be read, as opposed to specific columns
-- `exclude.na.listwise`: columns where missing values should be deleted listwise
+**Tip**: If you don't want the data to be automatically loaded, you have to set `preloadData: false` in the corresponding `qml` file.
 
-Note that we only read the data if we are ready for computing (as we verified in [Step 2 - Checking if Results can be Computed](#step-2---checking-if-results-can-be-computed)):
-
-<p><details>
-	<summary>Code</summary>
-
-  ```r
-  BinomialTest <- function(jaspResults, dataset, options) {
-
-    ready <- (length(options$variables) > 0)
-
-    if (ready)
-      dataset <- .binomReadData(dataset, options)
-  ```
-
-</details></p>
-
-Where `.binomReadData()` looks like:
-
-<p><details>
-	<summary>Code</summary>
-
-  ```r
-  .binomReadData <- function(dataset, options) {
-    if (!is.null(dataset)) # If data has been already read...
-      return(dataset)      # ... just return it ...
-    else                   # ... otherwise, read it
-      return(.readDataSetToEnd(columns.as.factor = options$variables))
-  }
-  ```
-
-</details></p>
+Reading data is always a tricky matter. Computers are devoid of common sense, so very often we have to explicitly specify how variables must be interpreted (are they a scale?, a nominal?, an ordinal?, ...). The `Edit data` panel allows you to specify this kind of things.
```

Comment:
We dont have to do this via edit data, you can define which type you want in each qml component.
The type in the data can stay as it is.

This used to be false, so im guessing that is where this text comes from

Reply:
```suggestion
Reading data is always a tricky matter. Computers are devoid of common sense, so very often we have to explicitly specify how variables must be interpreted (are they a scale?, a nominal?, an ordinal?, ...). Each column in the data has a type set on it, which is how it is shown in the dataviewer. However, for an analysis you might well want to get a different type, just for that analysis. Or even get a scale *and* a nominal version of the column.
The preloading of data in JASP actually takes that into account and adds every column selected in any qml component to a dataset, converted to exactly the type requested by the user.
Just like you can write `columnName.scale` or `columnName.nominal` in a JASP filter or computed column.
```